c10/core/TensorImpl.h

```diff
@@ -1578,8 +1603,13 @@ struct C10_API TensorImpl : public c10::intrusive_ptr_target {
 //    (optional) device
 //    miscellaneous bitfield
 //
+#ifdef NAMEDTENSOR_ENABLED
+#define NWORDS 30
+#else
+#define NWORDS 29
+#endif
 static_assert(sizeof(void*) != sizeof(int64_t) || // if 64-bit...
-              sizeof(TensorImpl) == sizeof(int64_t) * 29,
+              sizeof(TensorImpl) == sizeof(int64_t) * NWORDS,
```

Comment:
It is indeed only one word. This is because we're storing a `unique_ptr<NamedTensorMetaInterface>`, which you can think of as effectively a `NamedTensorMetaInterface*`. The size of a pointer is one word.

Reply:
Won't there still be one `NamedTensorMeta` allocation per tensor though? or I guess, only if names are specified.... ah OK. user can always set to nullptr and there's no extra memory, makes sense.

Reply:
There is one (or zero) `NamedTensorMeta` allocation per tensor. However, it's not reflected in `sizeof(TensorImpl)` because `TensorImpl` holds a **pointer** to `NamedTensorMeta` (as opposed to storing a NamedTensorMeta **value** as a field)